### DISCUSSION TITLE
Connection used by migrations loaded from elsewhere

### DISCUSSION THREAD

**Discussion by fabioluciano** [SeaQL/sea-orm#749] [Q&A] [answered]
## Motivation
I'm using SeaORM to create the databases to a CLI app. In the current implementation, the `DATABASE_URL` variable is loaded from `.env` file, using dotenv crate.

https://github.com/SeaQL/sea-orm/blob/e00562695ebb99eca444acbce41aafd0c5f3d539/sea-orm-migration/src/cli.rs#L11-L20

And, if i need to migrate the database, using, for example, a configuration file, other then .env, i cant.

## Proposed Solutions
Change the method signature to accept a connection, instead of load from dotenv directly.

Something like
```rust
pub async fn run_cli<M>(migrator: M, connection: DatabaseConnection)
where
    M: MigratorTrait,
{
    let app = build_cli();
    get_matches(migrator, &connection, app).await;
}
```

**Comment by fabioluciano** [comment]
I tried to implement the solution in a new [pull request](https://github.com/SeaQL/sea-orm/pull/738), but i dont know how to implement the connection in actix and in the other frameworks in the test scenarios.

**Comment by billy1624** [comment, marked as answer]
Hey @fabioluciano, thanks for the suggestion and contribution! If you want to run migration on startup, you can refer to our Actix examples. Basically, you can initialize the connection then pass it into methods in `MigratorTrait` to run the migrations. So actually you don't need the migration CLI.

https://github.com/SeaQL/sea-orm/blob/3f90c094079501211e88ea2111ce8f2473493bdf/examples/actix_example/src/main.rs#L173-L176

**Comment by fabioluciano** [comment]
Ow.... Let me try this 👯

**Comment by fabioluciano** [comment]
Ow! Worked! Thank you very much @billy1624

**Comment by billy1624** [comment]
Welcome! :D
